lib/std/c.zig

```diff
@@ -13,30 +13,24 @@ pub const tokenizer = @import("c/tokenizer.zig");
 pub const Token = tokenizer.Token;
 pub const Tokenizer = tokenizer.Tokenizer;
 
-/// The return type is `type` to force comptime function call execution.
-/// TODO: https://github.com/ziglang/zig/issues/425
-/// If not linking libc, returns struct{pub const ok = false;}
-/// If linking musl libc, returns struct{pub const ok = true;}
-/// If linking gnu libc (glibc), the `ok` value will be true if the target
+/// If not linking libc, returns false.
+/// If linking musl libc, returns true.
+/// If linking gnu libc (glibc), returns true if the target
 /// version is greater than or equal to `glibc_version`.
-/// If linking a libc other than these, returns `false`.
-pub fn versionCheck(comptime glibc_version: std.SemanticVersion) type {
-    return struct {
-        pub const ok = blk: {
-            if (!builtin.link_libc) break :blk false;
-            if (builtin.abi.isMusl()) break :blk true;
-            if (builtin.target.isGnuLibC()) {
-                const ver = builtin.os.version_range.linux.glibc;
-                const order = ver.order(glibc_version);
-                break :blk switch (order) {
-                    .gt, .eq => true,
-                    .lt => false,
-                };
-            } else {
-                break :blk false;
-            }
+/// If linking a libc other than these, returns false.
+pub fn versionCheck(comptime glibc_version: std.SemanticVersion) bool {
```

Comment:
This breaks the intended usage of the API which was that it would ensure called at comptime. Elsewhere you had to put `comptime` to make it still work correctly... I feel that this PR title is misleading, and this patch makes things worse.